modules/server_new/src/main/scala/observe/server/ObserveEngine.scala

```diff
@@ -558,50 +540,71 @@ object ObserveEngine {
           )
             .as(SeqEvent.NullSeqEvent)
         }.getOrElse(
-          Handle.pure[F, EngineState[F], Event[F, EngineState[F], SeqEvent], SeqEvent](NullSeqEvent)
+          EngineHandle.pure[F, SeqEvent](NullSeqEvent)
         )
       }
 
   private def tryAtomReload[F[_]: MonadThrow: Logger](
     odb:           OdbProxy[F],
     translator:    SeqTranslate[F],
-    executeEngine: Engine[F, EngineState[F], SeqEvent],
+    executeEngine: Engine[F],
     obsId:         Observation.Id,
     atomType:      SequenceType,
     onAtomReload:  OnAtomReloadAction
-  ): Handle[F, EngineState[F], EventType[F], Unit] =
-    Handle
-      .fromStream[F, EngineState[F], EventType[F]](Stream.eval {
-        Logger[F].debug(s"Reloading atom for observation [$obsId]") >>
-          odb
-            .read(obsId)
-            .map { odbObs =>
-              // Read the next atom from the odb and replaces the current atom
-              val atomGen: Option[AtomGen[F]] = translator.nextAtom(odbObs, atomType)._2
-              Event
-                .modifyState[F, EngineState[F], SeqEvent]({ (oldState: EngineState[F]) =>
+  ): EngineHandle[F, Unit] =
+    EngineHandle.fromSingleEventF {
+      Logger[F].debug(s"Reloading atom for observation [$obsId]") >>
+        odb
+          .read(obsId)
+          // TODO Simulate error.
```

Comment:
??? Why do we need to simulate an error here? Maybe this belongs to an instrument simulator?